src/main/java/org/mule/service/scheduler/internal/executor/ByCallerThreadGroupPolicy.java

```diff
@@ -61,15 +71,26 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
     ThreadGroup currentThreadGroup = currentThread().getThreadGroup();
 
     if (isWaitGroupThread(targetGroup) && targetGroup == currentThreadGroup) {
+      logRejection(r.toString(), "callerRuns", targetGroup.getName());
       callerRuns.rejectedExecution(r, executor);
     } else if (!isSchedulerThread(currentThreadGroup) || isWaitGroupThread(currentThreadGroup)) {
+      logRejection(r.toString(), "wait", targetGroup.getName());
       // MULE-11460 Make CPU-intensive pool a ForkJoinPool - keep the parallelism when waiting.
       wait.rejectedExecution(r, executor);
     } else {
+      logRejection(r.toString(), "abort", targetGroup.getName());
       abort.rejectedExecution(r, executor);
     }
   }
 
+  private void logRejection(String taskAsString, String strategy, String targetAsString) {
+    if (logRejectionDetails) {
+      LOGGER.warn("Task rejected ({}) from '{}' scheduler: {}", rightPad(strategy, 10), targetAsString, taskAsString);
+    } else if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("Task rejected ({}) from '{}' scheduler: {}", rightPad(strategy, 10), targetAsString, taskAsString);
```

Comment:
Can this be improved slightly.  I'm unsure if the taskAsString adds much value (at least in example), instead is there any way of output the class:line or a mini-stack of where the failed schedule occured.